cscs-checks/microbenchmarks/strided_bandwidth/strides.py

```diff
@@ -0,0 +1,88 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class StridedBase(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.sourcepath = 'strides.cpp'
+        self.build_system = 'SingleSource'
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'daint:mc', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.num_tasks = 0
+        self.num_tasks_per_node = 1
+
+        self.sanity_patterns = sn.assert_eq(
+            sn.count(sn.findall(r'bandwidth', self.stdout)),
+            self.num_tasks_assigned)
+
+        self.perf_patterns = {
+            'bandwidth': sn.extractsingle(
+                r'bandwidth: (?P<bw>\S+) GB/s',
+                self.stdout, 'bw', float)
+        }
+
+
+        self.maintainers = ['SK']
+        self.tags = {'benchmark', 'diagnostic'}
+
+    @property
+    @sn.sanity_function
+    def num_tasks_assigned(self):
+        return self.job.num_tasks
+
+
+@rfm.required_version('>=2.16-dev0')
+@rfm.simple_test
+class StridedBandwidthTest64(StridedBase):
+    def __init__(self):
+        super().__init__()
+
+        # 64-byte stride, using 1/8 of the cachline
+        self.executable_opts = ['100000000', '8', '1']
```

Comment:
I would use the maximum number of threads for each case (without hyperthreading), so that we can compare with the stride-1 case, which is the STREAM benchmark.

Reply:
The numbers are not comparable, because for STREAM, dest != src for the buffer, while here,
we have dest == src. Also the nature of the strided access precludes using vector instructions,
which is not the case for STREAM.
But I changed the test to using all threads, because all the bandwidth tests do the same and it will be easier to compare numbers.